streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -59,4 +89,55 @@ public SourceNode source(String topic) {
         return stateStoreSuppliers;
     }
 
+
+    private String childrenToString(List<ProcessorNode<?, ?>> children) {
+        if (children == null || children.isEmpty()) {
+            return null;
+        }
+
+        StringBuilder sb = new StringBuilder("children [");
+        for (ProcessorNode child : children) {
+            sb.append(child.name() + ",");
+        }
+        sb.setLength(sb.length() - 1);
+        sb.append("]\n");
+
+        // recursively print children
+        for (ProcessorNode child : children) {
```

Comment:
Could we just add a toString function to each of the ProcessorNode so that this code just becomes:

```
print(node) {

// node.toString
// also include listing children names

foreach(child <- children) print(child).
}
```